spec/index.adoc

```diff
@@ -415,6 +415,9 @@ tagged<t> = #6.55799(t) ; the CBOR tag
 +
 NOTE: Because this uses the lexicographic ordering of princpials, and the byte distinguishing the various classes of ids is at the _end_, this range by construction conceptually includes principals of various classes. This specification needs to take care that the fact that principals that are not canisters may appear in these ranges does not cause confusion.
 
+* `/subnet/<subnet_id>/nodes/<node_id>/public_key` (blob)
++
+The public key of a node (a DER-encoded Ed25519 signing key, see https://tools.ietf.org/html/rfc8410[RFC 8410] for reference) with principal `<node_id>` belonging to the subnet with principal `<subnet_id>`.
```

Comment:
Is that data on every subnet? Or only the root subnet? Or on each subnet, but only that subnet?

Reply:
But we still need to specify (precisely, ideally) what a client ought to check when validating a signature. How do they get the public key, and know whether its the right one? Should they trust any subnet (as long as it has a valid delegation), only the root subnet, or only the subnet of the current canister, or the latter two.

Reply:
Added a precise definition of query response signature validation to the spec.